lib/ansible/module_utils/basic.py

```diff
@@ -115,9 +121,9 @@
 # this matters, make sure to check for strings before checking for sequencetype
 try:
     from collections.abc import KeysView
-    SEQUENCETYPE = (Sequence, KeysView)
+    SEQUENCETYPE = (Sequence, KeysView, map)
```

Comment:
This is wrong as map is not a Sequence.  Pehaps we need to be checking for Iterables in some specific sections of code instead of Sequences?